src/abstractdataframe/abstractdataframe.jl

```diff
@@ -375,29 +375,40 @@ describe(df, stats = [:min, :max])
 ```
 
 """
-function StatsBase.describe(df::AbstractDataFrame; stats::Union{Symbol,AbstractVector{Symbol}} =
-                            [:mean, :min, :median, :max, :nunique, :nmissing, :eltype])
-    # Check that people don't specify the wrong fields.
+StatsBase.describe(df::AbstractDataFrame) = _describe(df, [:mean, :min, :median, 
+                                                     :max, :nunique, :nmissing, 
+                                                     :eltype])
+function StatsBase.describe(df::AbstractDataFrame, stats::Union{Symbol, Pair{Symbol}}...)
+    _describe(df, collect(stats))
+end
+
+function _describe(df::AbstractDataFrame, stats::Vector)   
+    predefined_funs = Symbol[s for s in stats if s isa Symbol] 
+
     allowed_fields = [:mean, :std, :min, :q25, :median, :q75,
                       :max, :nunique, :nmissing, :first, :last, :eltype]
-    if stats == :all
-        stats = allowed_fields
-    end
 
-    if stats isa Symbol
-        if !(stats in allowed_fields)
+    if predefined_funs == [:all]
+        predefined_funs = allowed_fields
+        i = findfirst(s -> s == :all, stats)
+        splice!(stats, i, allowed_fields) # insert in the stats vector to get a good order
```

Comment:
I need to also add an informative error when the user asks for `describe(df, :mean, :all)`.

Reply:
But this will be handled below. I have analyzed this case when reading the code 😄. Of course more informative message is welcome as now one would learn that `:all` is not allowed (and key is to describe all the options precisely in the docs).

If we are at it adding some failing tests would not hurt either.

Reply:
Is this fixed now?

Reply:
yes this is fixed.